File: Solutions/6. Invert Binary Tree/sol.cpp
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(root == NULL) return root;

        //swapping left and right child
        TreeNode* temp = root->left;
        root->left = root->right;
        root->right = temp;

        invertTree(root->left);
        invertTree(root->right);

        return root;
    }
};

File: Solutions/21. Diameter of Binary Tree/sol.java
class Solution {
    int dia = 0 ;
    public int diameterOfBinaryTree(TreeNode root) {
        maxheight(root) ;
        return dia ;
    }
    private int maxheight(TreeNode root) {
        if (root == null) 
            return 0;
    
        int left = maxheight(root.left);
        int right = maxheight(root.right);    
        dia = Math.max(dia , left + right) ;
        
        return Math.max(left, right) + 1;
    }
}
